src/lib.rs

```diff
@@ -667,6 +669,33 @@ decl_module! {
 
             Ok(())
         }
+
+        #[weight = FunctionOf(
+            |args: (&T::AccountId, &u8, &[u8; 32], &Timepoint<T::BlockNumber>)| 500_000 * (*args.1 as u32),
+            DispatchClass::Normal,
+            true
+        )]
+		fn clear(origin, multi_account_id: T::AccountId, max_cancellations: u8, call_hash: [u8; 32], timepoint: Timepoint<T::BlockNumber>) -> DispatchResult {
```

Comment:
I think we might need to set a static maximum upper bound, in case that the list is long enough that doesnt make it on time to fit the block time. Thoughts?

Reply:
I think, the maximum upper bound is 255, bounded by the data_type of `max_cancellations` which is u8.

Reply:
Yeah, indeed, I am thinking that 255 operations is assumable or not in 6 secs. I believe it should be enough to fit in a block, but would be great to see some perf metrics.

Reply:
Good point, maybe we want to use a bigger type and make the concrete size configurable through a trait type?